### PR TITLE
[FIX] themeSwitch class + [ADD] password component + ...

### PR DESCRIPTION
**Pull request by hamid-gh98**
- fix darkClass in themeSwitcher
- update sub remarks
- add password component
- update docker-compose
- fix and update some translations

### CODE REVIEW

File: web/assets/js/util/common.js [hunk 1]
```diff
@@ -69,48 +69,39 @@ function debounce(fn, delay) {
...
 }
-
-function areAllItemsExist(array1, array2) {
-    for (let i = 0; i < array1.length; i++) {
-      if (!array2.includes(array1[i])) {
-        return false;
-      }
-    }
-    return true;
-  }
```
Comment (alireza0) [inline web/assets/js/util/common.js:116]: It is about a conflict. Please fix it

File: web/html/xui/form/tls_settings.html
Comment (alireza0) [inline web/html/xui/form/tls_settings.html:1]: This one is updated in parallel. Please rebase it.

**Review by alireza0 — CHANGES_REQUESTED** [review]
Good job!
Please fix this minor changes due happened to parallel commits.

**Comment by hamid-gh98** [comment]
Hello @alireza0,
Done.

**Comment by hamid-gh98** [comment]
btw, when I go to settings.html, it says that `areAllItemsExist` is not defined.
so, I moved its function to `settings.html` and renamed it to `doAllItemsExist`.

image

**Comment by hamid-gh98** [comment]
I also now split the country configs (block + direct) in settings page.
hope that helps better

**Comment by alireza0** [comment]
> btw, when I go to settings.html, it says that `areAllItemsExist` is not defined. so, I moved its function to `settings.html` and renamed it to `doAllItemsExist`.
> 
> image

Actually it was defined [here](https://github.com/alireza0/x-ui/blob/main/web/assets/js/util/common.js#L109:L116) , but it seems in your branch it was not updated.
It is better to have general functions in `common.js` instead of directly push it to busy file like settings.
Please move `areAllItemsExist` or `doAllItemsExist` to its right place.
Overall your job is great! Thanks a lot

**Comment by hamid-gh98** [comment]
> > btw, when I go to settings.html, it says that `areAllItemsExist` is not defined. so, I moved its function to `settings.html` and renamed it to `doAllItemsExist`.
> 
> Actually it was defined [here](https://github.com/alireza0/x-ui/blob/main/web/assets/js/util/common.js#L109:L116) , but it seems in your branch it was not updated. It is better to have general functions in `common.js` instead of directly push it to busy file like settings. Please move `areAllItemsExist` or `doAllItemsExist` to its right place. Overall your job is great! Thanks a lot

ok, i removed it from `settings.html`.
thanks.

**Comment by alireza0** [comment]
Well done!

**Pull request merged**